Change to pre-defined colors by touching the legs of Clyde

jorainer opened the pull request:
- Touching a leg will change the ambient light to a pre-defined color; one color per leg; long touch will still trigger the color select.
- Saving the current ambient color when switching of the ambient light.
- Changed the sunset colors to red-to-blue transition like in a real sunset.

Commit:
+ Just changed the sunset colors to just represent reddish colors and
shortened the sunset period triggered by AfraidOfTheDark module.

Commit:
+ Touching a leg will change the color of the ambient light.
+ Changed the sunset colors to red-to-blue transition as in a real sunset.
+ Saving the current ambient color when switching of the ambient light.

Commit:
+ Fixed a problem with the sunset colors and Clyde not switching off the
light.
+ Added Time and TimeAlarms libraries, but not working...yet.

Commit:
+ Preventing color change by touching a leg during sunset.

Commit:
+ Prevent color select on leg touch during sunset.

Commit:
+ Added the Time and TimeAlarms libraries.
+ Added a ino that checks for an alarm.
+ Shortened the sunset period slightly.

metamanda commented:
Hi Johannes,

I love these changes!

I'm not sure we're ready to merge this into the master branch just yet. The beauty of Clyde is that people get to choose and change different behaviors. But I would love to tweet and send out links to your fork of the Clyde firmware so other backers can try it out. If we get positive feedback on it, then we can fold some of your features into version 2. Sound like a plan?

best,
Amanda

jorainer commented:
That's perfectly fine for me! I love your Clyde. I'm now also planning to include the Time and TimeAlarms library to use Clyde as an "alarm clock".

cheers, jo

metamanda commented:
Awesome! I'll get on The Twitter and mention the cool changes you made on
your fork then. :)

On Thu, Aug 21, 2014 at 2:39 PM, jotsetung notifications@github.com wrote:

> That's perfectly fine for me! I love your Clyde. I'm now also planning to
> include the Time and TimeAlarms library to use Clyde as an "alarm clock".
> 
> cheers, jo
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/fabule/Clyde/pull/7#issuecomment-52963454.

Commit:
Merge remote-tracking branch 'upstream/master' into timealarms

Commit:
+ Added a section regarding Time alarm settings in the README.md.
+ Some updates.

Commit:
+ Replaced the SerialCommand library shipped with Clyde with
the (smaller) one from Stefan Rado.
+ Added ENABLE_MOUTH to "hide" all sound output code if ENABLE_MOUTH is
not defined; otherwise the compiled code is too big to be uploaded...
+ Replaced some floats with uint8 to reduce the size of the firmware: in
the fadeWhite and fadeAmbient functions; the second parameter is now a
integer value with larger values causing longer, slower fades.
+ Defined example alarm functions in the ClydeTimeAlarms.ino firmware
that allow Clyde to be used as a alarm clock.

Commit:
+ Fixed the h3. header in the README.md.
+ Fixed a potential bug with Clyde changing the eye-color during sunset
or sunrise due to an accidental touch (real or false positive) of a leg.

Commit:
+ Also allowing cycle "UNKNOWN" to trigger color change on leg-touch.

Commit:
+ Some minor fixes.

Commit:
+ Included ENABLE_EYE preprocessor variable in Clyde.h that, if not
defined allows to use one of Clyde's legs as a switch (instead of the eye).

Commit:
+ Allow usage of Clyde without the eye.
